### ISSUE TITLE
New crash when closing and open the app

### ISSUE THREAD

**Issue by AlexaderDNT** [mapbox/mapbox-maps-ios#1752]
## Environment

- Xcode version: 14.1
- iOS version: 16.1.1
- Devices affected:
- Maps SDK Version: 10.9.1

![Screenshot 2022-11-23 at 12 34 08](https://user-images.githubusercontent.com/64775721/203536993-b8d8b8ec-878a-4cfa-9bc8-a1d89893f1b5.png)
![Screenshot 2022-11-23 at 12 51 38](https://user-images.githubusercontent.com/64775721/203540215-2708f71c-775e-4965-806a-5d622fb03e63.png)

**Comment by ZiZasaurus** [comment]
@AlexaderDNT can you please provide some additional information so we can investigate this issue:

- What is happening in the app when this crash occurs?
- Can you provide us with a screenshot of your stack trace?
- Can you provide us with a sample app that reproduces this behavior?

**Comment by AlexaderDNT** [comment]
It happens when i run the app.

Stack trace
Fatal Exception: Bindgen internal error
0  CoreFoundation                 0x9e88 __exceptionPreprocess
1  libobjc.A.dylib                0x178d8 objc_exception_throw
2  CoreFoundation                 0xf9c58 __CFDictionaryCreateGeneric
3  MapboxCoreMaps                 0x3a738 (Missing UUID 6ac32b716dd5349b9a4232abd9d8337f)
4  UT                             0x176be70 @nonobjc MBMStyleManager.setStyleLayerPropertiesForLayerId(_:properties:) (<compiler-generated>)
5  UT                             0x176bda0 protocol witness for StyleManagerProtocol.setStyleLayerPropertiesForLayerId(_:properties:) in conformance MBMStyleManager (<compiler-generated>)
6  UT                             0x176166c closure #1 in Style.setLayerProperties(for:properties:) + 542 (Style.swift:542)
7  UT                             0x175ca28 Style.handleExpected<A, B>(closure:) + 1080 (Style.swift:1080)
8  UT                             0x175e158 Style.setLayerProperties(for:properties:) + 541 (Style.swift:541)
9  UT                             0x1765890 protocol witness for StyleProtocol.setLayerProperties(for:properties:) in conformance Style (<compiler-generated>)
10 UT                             0x1634e60 Puck2D.updateLayerLocationFastPath() + 250 (Puck2D.swift:250)
11 UT                             0x1634838 Puck2D.latestLocation.didset + 51 (Puck2D.swift:51)
12 UT                             0x1632f2c Puck2D.latestLocation.setter + 47 (Puck2D.swift:47)
13 UT                             0x1632e68 closure #1 in Puck2D.isActive.didset + 19 (Puck2D.swift:19)
14 UT                             0x1631370 thunk for @escaping @callee_guaranteed (@unowned InterpolatedLocation) -> (@unowned Bool) (<compiler-generated>)
15 UT                             0x1606a7c closure #1 in ObservableValue.observe(with:) + 27 (ObservableValue.swift:27)
16 UT                             0x1606568 ObservableValue.Observer.invokeHandler(with:) + 52 (ObservableValue.swift:52)
17 UT                             0x1606494 closure #1 in ObservableValue.notify(with:) + 20 (ObservableValue.swift:20)
18 UT                             0x16064d0 partial apply for closure #1 in ObservableValue.notify(with:) (<compiler-generated>)
19 libswiftCore.dylib             0xe53b8 Sequence.forEach(_:)
20 UT                             0x1606404 ObservableValue.notify(with:) + 19 (ObservableValue.swift:19)
21 UT                             0x1631250 ObservableInterpolatedLocation.notify(with:) + 21 (ObservableInterpolatedLocation.swift:21)
22 UT                             0x1631e88 protocol witness for ObservableInterpolatedLocationProtocol.notify(with:) in conformance ObservableInterpolatedLocation (<compiler-generated>)
23 UT                             0x1625f2c InterpolatedLocationProducer.participate() + 146 (InterpolatedLocationProducer.swift:146)
24 UT                             0x1625f58 protocol witness for DisplayLinkParticipant.participate() in conformance InterpolatedLocationProducer (<compiler-generated>)
25 UT                             0x15f3384 MapView.updateFromDisplayLink(displayLink:) + 574 (MapView.swift:574)
26 UT                             0x15f3f70 closure #1 in MapView.didMoveToWindow() + 619 (MapView.swift:619)
27 UT                             0x15de414 ForwardingDisplayLinkTarget.update(with:) + 12 (ForwardingDispalyLinkTarget.swift:12)
28 UT                             0x15de460 @objc ForwardingDisplayLinkTarget.update(with:) (<compiler-generated>)
29 QuartzCore                     0x28f9c CA::Display::DisplayLink::dispatch_items(unsigned long long, unsigned long long, unsigned long long)
30 QuartzCore                     0x155edc CA::Display::DisplayLink::dispatch_deferred_display_links(unsigned int)
31 UIKitCore                      0x652740 _UIUpdateSequenceRun
32 UIKitCore                      0xc99fd0 schedulerStepScheduledMainSection
33 UIKitCore                      0xc9919c runloopSourceCallback
34 CoreFoundation                 0xd5f54 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__
35 CoreFoundation                 0xe232c __CFRunLoopDoSource0
36 CoreFoundation                 0x66210 __CFRunLoopDoSources0
37 CoreFoundation                 0x7bba8 __CFRunLoopRun
38 CoreFoundation                 0x80ed4 CFRunLoopRunSpecific
39 GraphicsServices               0x1368 GSEventRunModal
40 UIKitCore                      0x3a23d0 -[UIApplication _run]
41 UIKitCore                      0x3a2034 UIApplicationMain
42 SwiftUI                        0x1d1014 OUTLINED_FUNCTION_895
43 SwiftUI                        0x13216c block_copy_helper.1
44 SwiftUI                        0x11b4bc OUTLINED_FUNCTION_901
45 UT                             0x396784 static UTApp.$main() (<compiler-generated>)
46 UT                             0x39682c main (UTApp.swift)
47 ???                            0x1b63fc960 (Missing)

**Comment by Taeeun-Kim** [comment]
with version 5.9.0 <- no problem
MapboxMaps SDK with version 10.9.0/10.9.1 using **Pod** <- no problem
MapboxMaps SDK with version 10.9.0/10.9.1 using **SPM** <- I'm also having the same issue.

**Comment by AlexaderDNT** [comment]
Fixed some of my crashes now by removing layers. I have a lot of features, 20 000+ and i needed to split them up in many layers, i had like 40-50 layers with 2000 each layer. I did this because i did filter on id, and there is a limit when you filter on layer with many features. So i removed 40+ layers and now i am only filtering on tags instead of id, the bug is gone but i still got a issue on older iphone where memory is up to 1.3 gb, iphone 6s crashes when it comes up to 1.4. I can share this crash in another thread when i find it.
